### PR TITLE
cleanup karmada ctl code

### PR DESCRIPTION
**Pull request by gy95**
Signed-off-by: guoyao <1015105054@qq.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
structure `CommandGetOptions` has some un-used fields, we can remove these.
karmadactl get/cordon cmds not clean, modify some logics.
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/karmadactl/cordon.go [hunk 1]
```diff
@@ -93,15 +93,8 @@ func NewCmdUncordon(cmdOut io.Writer, karmadaConfig KarmadaConfig, cmdStr string
...
-	KubeConfig string
-
-	// ClusterContext is the name of the cluster context in control plane KUBECONFIG file.
-	// Default value is the current-context.
-	KarmadaContext string
-
-	// DryRun tells if run the command in dry-run mode, without making any server requests.
-	DryRun bool
+	// global flags
+	options.GlobalCommandOptions
```
Comment (lonelyCZ) [inline pkg/karmadactl/cordon.go:97]: There is a flag `ClusterNamespace` in GlobalCommandOptions that is not used by this command.

I consider whether remove it from GlobalCommandOptions and only remain the most universal flag in GlobalCommandOptions.

Comment (RainbowMango) [reply pkg/karmadactl/cordon.go:97]: The #1011 has been merged, shall we rebase this?

File: pkg/karmadactl/get.go [hunk 2]
```diff
@@ -123,62 +119,64 @@ type CommandGetOptions struct {
...
-// Complete takes the command arguments and infers any remaining options.
-func (g *CommandGetOptions) Complete(cmd *cobra.Command, args []string) error {
+func NewCommandGetOptions() *CommandGetOptions {
 	newScheme := gclient.NewSchema()
-	// human readable printers have special conversion rules, so we determine if we're using one.
-	g.IsHumanReadablePrinter = true
+	kubeConfig := defaultKubeConfig
+	printFlags := get.NewGetPrintFlags()
+
+	ioStreams := genericclioptions.IOStreams{In: getIn, Out: getOut, ErrOut: getErr}
```
Comment (lonelyCZ) [inline pkg/karmadactl/get.go:128]: I suggest `NewCommandGetOptions` should be as brief as possible, more logic of initialization should be set in `Complete`.

This code logic comes from k8s, I advise not to change it.

File: pkg/karmadactl/get.go [hunk 3]
```diff
@@ -123,62 +119,64 @@ type CommandGetOptions struct {
...
 
-	// check karmada config path
+	// using env variable if not empty
 	env := os.Getenv("KUBECONFIG")
 	if env != "" {
-		g.KubeConfig = env
+		kubeConfig = env
```
Comment (lonelyCZ) [inline pkg/karmadactl/get.go:132]: Perhaps, we don't need to check env `KUBECONFIG`,  none of other subcommands check this env, but they can run. So I think we can optimize it here.

**Comment by lonelyCZ** [comment]
Hi, @QAQ-rookie 
I want to invite you to review this pr, the below fields not to be used effectively in `karmadactl get`, can we delete them? Or whether to consider extending the contents of these fields in the future?

https://github.com/karmada-io/karmada/blob/668eaee05b208d80e30957c9440c3ad6def69a2a/pkg/karmadactl/get.go#L97-L98

https://github.com/karmada-io/karmada/blob/668eaee05b208d80e30957c9440c3ad6def69a2a/pkg/karmadactl/get.go#L115

**Comment by QAQ-rookie** [comment]
These fields can be deleted.



------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Zhe ***@***.***&gt;; 
发送时间: 2021年11月29日(星期一) 晚上8:45
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [karmada-io/karmada] cleanup karmada ctl code (PR #946)





 
Hi, @QAQ-rookie
 I want to invite you to review this pr, the below fields not to be used effectively in karmadactl get, can we delete them? Or whether to consider extending the contents of these fields in the future?
 
https://github.com/karmada-io/karmada/blob/668eaee05b208d80e30957c9440c3ad6def69a2a/pkg/karmadactl/get.go#L97-L98
 
https://github.com/karmada-io/karmada/blob/668eaee05b208d80e30957c9440c3ad6def69a2a/pkg/karmadactl/get.go#L115

**Comment by lonelyCZ** [comment]
@QAQ-rookie Thank you!

**Comment by lonelyCZ** [comment]
Hi, @gy95 
There seem some errors?
```
[root@master67 karmada]# ./karmadactl get pod
NAME   AGE
       <unknown>
```

**Comment by gy95** [comment]
> Hi, @gy95 There seem some errors?
> 
> ```
> [root@master67 karmada]# ./karmadactl get pod
> NAME   AGE
>        <unknown>
> ```

fixed. :)

File: pkg/karmadactl/get.go (hunk 3 above)
Comment (gy95) [reply pkg/karmadactl/get.go]: this is need, or we'll need to emphasize the kubeconfig when executing the command, like below.
./karmadactl get ns --kubeconfig=/root/.kube/karmada.config

**Comment by lonelyCZ** [comment]
thanks, @gy95 

/lgtm

**Pull request merged**